storage_service/locations/models/package.py

```diff
@@ -1018,14 +1027,31 @@ def finish_reingest(self, origin_location, origin_path, reingest_location, reing
         LOGGER.debug('Reingest AIP full path: %s', reingest_full_path)
 
         # Copy pointer file if exists
-        # TODO what do if LOCKSS & uncompressed?
-        if self.package_type in (Package.AIP, Package.AIC):
+        # TODO what do if LOCKSS?
+        if self.package_type in (Package.AIP, Package.AIC) and to_be_compressed:
             reingest_pointer_src = os.path.join(origin_location.relative_path, os.path.dirname(origin_path), 'pointer.xml')
-            reingest_pointer_name = 'pointer.' + self.uuid + '.reingest.xml'
-            reingest_pointer_dst = os.path.join(ss_internal.relative_path, reingest_pointer_name)
+
+            # If reingesting a previously compressed AIP, make a temporary "reingest" pointer (otherwise make a normal one)
+            if was_compressed:
+                reingest_pointer_name = 'pointer.' + self.uuid + '.reingest.xml'
+                reingest_pointer_dst = os.path.join(ss_internal.relative_path, reingest_pointer_name)
+            else:
+                reingest_pointer_name = 'pointer.' + self.uuid + '.xml'
+                reingest_pointer_dst = os.path.join(ss_internal.relative_path, utils.uuid_to_path(self.uuid), reingest_pointer_name)
```

Comment:
This pointer gets created at this location because the AIP doesn't yet have a pointer stored at that location and the update_pointer_file function needs a pointer file to update.